### ISSUE TITLE
NA values are shown as 0

### ISSUE THREAD

**Issue by friedue** [broadinstitute/single_cell_portal#12]
In the expression matrix that we uploaded we had numerous NA entries. We noticed that these are replaced with zero when you're coloring cells within the tSNE maps according to the expression of a given gene.

In principle this makes sense, however, since the expression matrix that we'd like to use for display contains scaled and log transformed counts, these NA should no longer be treated as zeros. Instead, there should be a way to indicate that the information is missing, e.g. by just displaying them as empty points (black line, no fill).

**Comment by TimothyTickle** [comment]
Hey Friedue,

Thanks for the comment, we did not expect NAs in an expression matrix, this would be uncommon. People tend to use log transformed + 1 values to avoid NAs. If you feel it is reasonable we could check for NAs for you on upload and fail the upload if that helps catch NAs for you.

What are your thoughts here?

**Comment by friedue** [comment]
I put the NAs there because I _wanted_ to have a distinction for the no-count genes. We also use an offset, so I guess I can just refrain from adding the NAs back in, but I tend to prefer to see a difference between very lowly expressed genes and non-detectable genes.

**Comment by friedue** [comment]
Just to give a very brief visual cue as to why I prefer to simply ignore the values from a given gene for a specific cell:

![image](https://cloud.githubusercontent.com/assets/4973546/25459277/8a8ccbd6-2aab-11e7-84e8-a9cc4c04d0fb.png)

Here, all the values that align with zero are, in fact, cases where we didn't detect any counts. This is because our NAs got replaced with zero.
Even if we had kept them in -- since we have an off-set of 0.01, they would have been assigned to -2, which, in turn, would have pulled the mean expression significantly down. I'd rather just compare the values in which I have some degree of confidence (with zeros you never really know why you got them)
